### PR TITLE
Fixing issue 1493-Change log type for missing key errors from 'error' to 'debug' + other logging improvements (issue 1484)

### PR DESCRIPTION
**Pull request by jessiejuachon**

### CODE REVIEW

**Review by Xiaochao8 — APPROVED** [review]

File: lib/sgtn-client/api/translation.rb [hunk 1]
```diff
@@ -12,12 +12,12 @@ module Core
...
 
       def self.getString(component, key, locale)
-        SgtnClient.logger.debug "[Translation][getString_p]component=#{component}, key=#{key}, locale=#{locale}"
+        SgtnClient.logger.debug "[Translation][getString]component=#{component}, key=#{key}, locale=#{locale}"
         str = getTranslation(component, key, locale)
         if str.nil?
           str = SgtnClient::Source.getSource(component, key, SgtnClient::Config.configurations.default)
           if str.nil?
-            SgtnClient.logger.error "[Translation]can't find the key '" + key + "' in source path!"
+            SgtnClient.logger.debug "[Translation.getString] Missing source string with key: #{key}, component: #{component}, locale=#{locale}"
```
Comment (Xiaochao8) [inline lib/sgtn-client/api/translation.rb:20]:
```suggestion
            SgtnClient.logger.debug "[Translation.getString] Missing source string with key: #{key}, component: #{component}, locale: #{locale}"
```

File: lib/sgtn-client/api/translation.rb [hunk 2]
```diff
@@ -12,12 +12,12 @@ module Core
   class Translation
 
       def self.getString(component, key, locale)
-        SgtnClient.logger.debug "[Translation][getString_p]component=#{component}, key=#{key}, locale=#{locale}"
+        SgtnClient.logger.debug "[Translation][getString]component=#{component}, key=#{key}, locale=#{locale}"
```
Comment (Xiaochao8) [inline lib/sgtn-client/api/translation.rb:15]:
```suggestion
        SgtnClient.logger.debug "[Translation.getString]component: #{component}, key: #{key}, locale: #{locale}"
```

File: lib/sgtn-client/api/translation.rb [hunk 3]
```diff
@@ -29,7 +29,7 @@ def self.getString_p(component, key, plural_args, locale)
         if str.nil?
           str = SgtnClient::Source.getSource(component, key, SgtnClient::Config.configurations.default)
           if str.nil?
-            SgtnClient.logger.error "[Translation]can't find the key '" + key + "' in source path!"
+            SgtnClient.logger.debug "[Translation.getString_p] Missing source string with key: #{key}, component: #{component}, locale=#{locale}"
```
Comment (Xiaochao8) [inline lib/sgtn-client/api/translation.rb:32]:
```suggestion
            SgtnClient.logger.debug "[Translation.getString_p] Missing source string with key: #{key}, component: #{component}, locale: #{locale}"
```

File: lib/sgtn-client/api/translation.rb [hunk 4]
```diff
@@ -62,7 +62,7 @@ def self.getStrings(component, locale)
           items = {}
           s = SgtnClient::Source.getSources(component, default)
           if s.nil?
-            SgtnClient.logger.error "[Translation]can't find the component '" + component + "' in source path!"
+            SgtnClient.logger.error "[Translation.getStrings] Missing source string with key: #{key}, component: #{component}, locale=#{locale}"
```
Comment (Xiaochao8) [inline lib/sgtn-client/api/translation.rb:65]:
```suggestion
            SgtnClient.logger.error "[Translation.getStrings] Missing source string with key: #{key}, component: #{component}, locale: #{locale}"
```

**Review by Xiaochao8 — APPROVED** [review]

File: lib/sgtn-client/api/translation.rb [hunk 5]
```diff
@@ -62,7 +62,7 @@ def self.getStrings(component, locale)
           items = {}
           s = SgtnClient::Source.getSources(component, default)
           if s.nil?
-            SgtnClient.logger.error "[Translation]can't find the component '" + component + "' in source path!"
+            SgtnClient.logger.error "[Translation][getStrings] Missing component: #{component}, locale: #{locale}"
```
Comment (Xiaochao8) [inline lib/sgtn-client/api/translation.rb:65]:
```suggestion
            SgtnClient.logger.error "[Translation.getStrings] Missing component: #{component}, locale: #{locale}"
```

File: lib/sgtn-client/api/translation.rb [hunk 6]
```diff
@@ -12,12 +12,12 @@ module Core
   class Translation
 
       def self.getString(component, key, locale)
-        SgtnClient.logger.debug "[Translation][getString_p]component=#{component}, key=#{key}, locale=#{locale}"
+        SgtnClient.logger.debug "[Translation.getString]component: #{component}, key: #{key}, locale: #{locale}"
```
Comment (Xiaochao8) [inline lib/sgtn-client/api/translation.rb:15]:
```suggestion
        SgtnClient.logger.debug class.name { "method: #{__method__}, key: #{key}, component: #{component}, locale: #{locale}" }
```

Comment (linr211) [reply lib/sgtn-client/api/translation.rb:15]: We could have another PR to make the change Xiaochao suggested globally.

**Review by linr211 — APPROVED** [review]

**Review by Xiaochao8 — APPROVED** [review]

File: lib/sgtn-client/api/translation.rb (hunk 6 above)
Comment (jessiejuachon) [reply lib/sgtn-client/api/translation.rb:15]: @Xiaochao8 , @linr211 To track this: https://github.com/vmware/singleton/issues/1512

**Pull request merged**